request/request.go

```diff
@@ -1,65 +1,148 @@
 	return Request{bridge.New(ch)}
 }
 
+// kong.Request.GetScheme() returns the scheme component of the request’s URL.
+// The returned value is normalized to lower-case form.
 func (r Request) GetScheme() (s string, err error) {
 	return r.AskString(`kong.request.get_scheme`)
 }
 
+// kong.Request.GetHost() returns the host component of the request’s URL,
+// or the value of the “Host” header. The returned value is normalized
+// to lower-case form.
 func (r Request) GetHost() (host string, err error) {
 	return r.AskString(`kong.request.get_host`)
 }
 
+// kong.Request.GetPort() returns the port component of the request’s URL.
 func (r Request) GetPort() (port string, err error) {
 	return r.AskString(`kong.request.get_port`)
 }
 
+// kong.Request.GetForwardedScheme() returns the scheme component
+// of the request’s URL, but also considers X-Forwarded-Proto if it
+// comes from a trusted source. The returned value is normalized to lower-case.
+//
+// Whether this function considers X-Forwarded-Proto or not depends
+// on several Kong configuration parameters:
+//
+//   - trusted_ips
+//   - real_ip_header
+//   - real_ip_recursive
+//
+// Note: support for the Forwarded HTTP Extension (RFC 7239) is not offered yet
+// since it is not supported by ngx_http_realip_module.
 func (r Request) GetForwardedScheme() (s string, err error) {
 	return r.AskString(`kong.request.get_forwarded_scheme`)
 }
 
+// kong.Request.GetForwardedHost() returns the host component of the request’s URL
+// or the value of the “host” header. Unlike kong.Request.GetHost(), this function
+// will also consider X-Forwarded-Host if it comes from a trusted source.
+// The returned value is normalized to lower-case.
+//
+// Whether this function considers X-Forwarded-Proto or not depends
+// on several Kong configuration parameters:
+//
+//   - trusted_ips
+//   - real_ip_header
+//   - real_ip_recursive
+//
+// Note: we do not currently offer support for Forwarded HTTP Extension (RFC 7239)
+// since it is not supported by ngx_http_realip_module.
 func (r Request) GetForwardedHost() (host string, err error) {
 	return r.AskString(`kong.request.get_forwarded_host`)
 }
 
+// kong.Request.GetForwardedPort() returns the port component of the request’s URL,
+// but also considers X-Forwarded-Host if it comes from a trusted source.
+//
+// Whether this function considers X-Forwarded-Proto or not depends
+// on several Kong configuration parameters:
+//
+//   - trusted_ips
+//   - real_ip_header
+//   - real_ip_recursive
+//
+// Note: we do not currently offer support for Forwarded HTTP Extension (RFC 7239)
+// since it is not supported by ngx_http_realip_module.
 func (r Request) GetForwardedPort() (port string, err error) {
 	return r.AskString(`kong.request.get_forwarded_port`)
 }
 
+// kong.Request.GetHttpVersion() returns the HTTP version
+// used by the client in the request, returning values
+// such as "1"", "1.1", "2.0", or nil for unrecognized values.
```

Comment:
same string / number divergence here.    and does it return an error on weird values?